test/source/coiteration_test.cpp

```diff
@@ -193,3 +196,74 @@ TEST_CASE("Coiteration-Singleton-Singleton-Dense-Dense")
     }
     CHECK(fn(std::tuple(it1 == end1, it2 == end2, it3 == end3, it4 == end4)) == true);
 }
+
+TEST_CASE("Coiteration-Dense-Hashed-ConjunctiveMerge")
+{
+    constexpr uint8_t ZERO = 0;
+
+    std::unordered_map<uintptr_t, uintptr_t> const umap1{ { 0, 1 }, { 2, 0 }, { 1, 2 } };
+    std::vector<std::unordered_map<uintptr_t, uintptr_t>> const crd0{ umap1 };
+
+    // initialize the two levels to be coiterated
+    xsparse::levels::dense<std::tuple<>, uintptr_t, uintptr_t> s1{ 5 };
+    xsparse::levels::hashed<
+        std::tuple<>,
+        uintptr_t,
+        uintptr_t,
+        xsparse::util::container_traits<std::vector, std::set, std::unordered_map>,
+        xsparse::level_properties<false, false, false, false, false>>
+        h { 5, crd0 };
+
+    // define a conjunctive function
+    auto fn = [](std::tuple<bool, bool> t) constexpr
+    {
+        return (std::get<0>(t) && std::get<1>(t));
+    };
+
+    xsparse::level_capabilities::Coiterate<std::function<bool(std::tuple<bool, bool>)>,
+                                           uintptr_t,
+                                           uintptr_t,
+                                           std::tuple<decltype(s1), decltype(h)>,
+                                           std::tuple<>>
+        coiter(fn, s1, h);
+
+    auto it_helper1 = s1.iter_helper(std::make_tuple(), ZERO);
+    auto it1 = it_helper1.begin();
+    auto it_helper2 = h.iter_helper(ZERO);
+    auto it2 = it_helper2.begin();
+
+    auto end1 = it_helper1.end();
+    auto end2 = it_helper2.end();
+
+    // when co-iterating over levels that are unordered (i.e. hashed), then we use locate to check
+    // if the index exists in the hashed level. If not, then we skip it.
+    for (auto const [ik, pk_tuple] : coiter.coiter_helper(std::make_tuple(), ZERO))
+    {
+        // get the index and pointer from the levels involved in co-iteration
+        auto [i1, p1] = *it1;
+        auto [i2, p2] = *it2;
```

Comment:
Better not to iterate over an unordered collection. Just insert into and check the contents of a vector.